src/strict_fstream.hpp

```diff
@@ -16,42 +15,41 @@
  */
 namespace strict_fstream
 {
-/// Overload of error-reporting function, to enable use with VS (1)
-/// and POSIX signature found in MUSL on Alpine (2)
-/// Ref 1: http://stackoverflow.com/a/901316/717706
-/// Ref 2: http://stackoverflow.com/a/41956165
-static inline char* strerror_r_compat(int result, char* buffer, int err)
-{
-    if (result)
-    {
-        sprintf(buffer, "Unknown error: %d", err);
-    }
-    return buffer;
-}
 
-static inline char* strerror_r_compat(char* result, char*, int)
-{
-    return result;
-}
+// Workaround for broken musl implementation
+// Since musl insists that they are perfectly compatible, ironically enough,
+// they don't have a __musl__ or similar. But __NEED_size_t is defined in their
+// relevant header (and not in working implementations), so we can use that.
+#ifdef __NEED_size_t
+#define BROKEN_GNU_STRERROR_R
+#warning "Working around broken strerror_r() implementation in musl, remove when musl is fixed"
+#endif
 
+/// Overload of error-reporting function, to enable use with VS.
+/// Ref: http://stackoverflow.com/a/901316/717706
 static std::string strerror()
 {
-    std::string buff(80, '\0');
+    std::string buff(256, '\0');
 #ifdef _WIN32
-    if (strerror_s(&buff[0], buff.size(), errno) != 0)
-    {
+    if (strerror_s(buff.data(), buff.size(), errno) == 0) {
+        buff.resize(buff.find('\0'));
+    } else {
         buff = "Unknown error";
     }
-#else
-    auto p = strerror_r_compat(strerror_r(errno, &buff[0], buff.size()), &buff[0], errno);
-    std::string tmp(p, std::strlen(p));
-    std::swap(buff, tmp);
-#endif
-    if(buff.find('\0') != std::string::npos)
-    {
+    return buff;
+#elif ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600 || defined(__APPLE__)) && ! _GNU_SOURCE) || defined(BROKEN_GNU_STRERROR_R)
+// XSI-compliant strerror_r()
+    if (strerror_r(errno, buff.data(), buff.size()) == 0) {
         buff.resize(buff.find('\0'));
```

Comment:
Same comment as above.

Reply:
I used ` [...]` instead, `WARNING: truncation` could be a bit confusing and I think everyone understands that `[...]` means that there's more output not displayed.